src/main/kotlin/FunCommands.kt

```diff
@@ -0,0 +1,109 @@
+import com.velocitypowered.api.command.SimpleCommand
+import com.velocitypowered.api.proxy.Player
+import com.velocitypowered.api.proxy.ProxyServer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import net.kyori.adventure.text.Component
+import org.slf4j.Logger
+import java.nio.file.Files
+import java.nio.file.Paths
+
+@Serializable
+data class FunCommandConfig(
+    val command: String,
+    val description: String,
+    val localChat: String? = null, // Add localChat as optional
+    val globalChat: String? = null, // Keep globalChat as optional
+    val run: String? = null // Optional execution logic
+)
+
+class FunCommands(
+    private val proxy: ProxyServer,
+    private val logger: Logger,
+    private val chattORE: ChattORE
+) {
+
+    fun loadFunCommands() {
+        val resourcePath = "commands.json"
+        val resourceStream = javaClass.classLoader.getResourceAsStream(resourcePath)
+
+        if (resourceStream == null) {
+            logger.warn("$resourcePath not found. Skipping fun command loading.")
+            return
+        }
+
+        val commandsJson = resourceStream.bufferedReader().use { it.readText() }
+        val commands = Json.decodeFromString<List<FunCommandConfig>>(commandsJson)
+        logger.info("Parsed ${commands.size} commands from JSON.")
+
+        commands.forEach { commandConfig ->
+            val meta = proxy.commandManager.metaBuilder(commandConfig.command).build()
+            proxy.commandManager.register(meta, createDynamicCommand(commandConfig))
+        }
+
+        logger.info("Loaded ${commands.size} fun commands.")
+    }
+
+    private fun createDynamicCommand(commandConfig: FunCommandConfig): SimpleCommand {
+        return SimpleCommand { invocation ->
+            val source = invocation.source()
+            val args = invocation.arguments()
+
+            if (source !is Player) {
+                source.sendMessage(Component.text("This command can only be used by players!"))
+                return@SimpleCommand
+            }
+
+            val player = source
+            val rawMessage = commandConfig.globalChat ?: commandConfig.localChat ?: ""
+
+            val filledMessage = rawMessage
+                .replace("<name>", player.username)
+                .replace("<arg-all>", args.joinToString(" "))
+                .replace("$1", args.getOrNull(1) ?: "<missing>")
+                .replace("$2", args.getOrNull(2) ?: "<missing>")
+
+            // Render the message using chattORE's render function
+            val renderedMessage = filledMessage.render(
+                mapOf(
+                    "message" to filledMessage.toComponent(),
+                    "sender" to player.username.toComponent()
+                )
+            )
+
+            if (commandConfig.globalChat != null) {
+                chattORE.broadcast(renderedMessage)
+            } else if (commandConfig.localChat != null) {
```

Comment:
I think it'd be nice if it could do both local and global chat if they were set, not one or the other. [The old FungeeCommands allowed for both to occur at the same time](https://github.com/OpenRedstoneEngineers/FungeeCommands/blob/f34c6d97d892c077f82ef19402f9dd65bbeae8ea/fungee-manager/src/main/kotlin/org/openredstone/fungee/manager/commands/GenericCommand.kt#L29).